File: contest/2022-10-12 (2)/资料/10-02/Src/wzj/DerProze.cpp
#include<stdio.h>
#include<stdlib.h>
long a,b,c,i,j,k,l,x,xq,y,z;
int main(){
	freopen("DerProze.in","r",stdin);
	freopen("DerProze.out","w",stdout);
	scanf("%ld%ld%ld",&a,&b,&c); 
	xq=6;
	if(a<2000){x=a;y=2000;z=1;}
	else{x=2000;y=a;z=-1;}
	for(i=x;i<y;i++){
		if((i%100!=0 && i%4==0)|| i%400==0){
			xq-=2*z;
			if(xq<1)xq+=7;
			if(xq>7)xq-=7;
		}else{
			xq-=1*z;
			if(xq<1)xq+=7;
			if(xq>7)xq-=7;
		}		
	}
	for(i=1;i<b;i++){
		if(i==4 || i==6 || i==9 || i==11 ){
			xq+=2;
			if(xq>7)xq-=7;
		}else if(i==2){
			if((a%100!=0 && a%4==0)|| a%400==0){
				xq+=1;
			}
			if(xq>7)xq-=7;
		}else{
			xq+=3;
			if(xq>7)xq-=7;
		}
	}
	xq+=c-1;
	xq=(xq-1)%7+1;
	if(xq==1)printf("monday\n");
	if(xq==2)printf("tuesday\n");
	if(xq==3)printf("wednesday\n");
	if(xq==4)printf("thursday\n");	
	if(xq==5)printf("friday\n");
	if(xq==6)printf("saturday\n");
	if(xq==7)printf("sunday\n");		
	return 0;
}// monday tuesday wednesday thursday friday  saturday sunday 

File: contest/2022-10-12 (2)/资料/10-02/Src/wym/DerProze.pas
var yyyy,mm,dd,y,m,d:integer;
	k:longint;
begin
	assign(input,'DerProze.in');reset(input);
	assign(output,'DerProze.out');rewrite(output);
	readln(yyyy,mm,dd);
	y:=1;m:=1;d:=1;k:=0;
	while not((y=yyyy)and(m=mm)and(d=dd)) do begin
		inc(d);inc(k);k:=k mod 7;
		if (d=32)and((m=1)or(m=3)or(m=5)or(m=7)or(m=8)or(m=10)or(m=12)) then begin
			inc(m);d:=1;
		end else if (d=31)and((m=4)or(m=6)or(m=9)or(m=11)) then begin
			inc(m);d:=1;
		end else if (m=2)then 
			if (((y mod 4=0)and not((y mod 100=0)xor(y mod 400=0)))and(d=30)) or 
				(not ((y mod 4=0)and not((y mod 100=0)xor(y mod 400=0)))and(d=29)) then begin
					inc(m);d:=1;
				end;
		if m=13 then begin inc(y);m:=1;end;
	end;
	case k of
		2:writeln('wednesday');
		3:writeln('thursday');
		4:writeln('friday');
		5:writeln('saturday');
		6:writeln('sunday');
		0:writeln('monday');
		1:writeln('tuesday');
		else writeln(k);
	end;
	close(input);close(output);
end.